core/sql/common/ComSmallDefs.h

```diff
@@ -609,47 +609,59 @@ enum ComColumnClass { COM_UNKNOWN_CLASS
 #define COM_ALTERED_USER_COLUMN_LIT         "C "
 
 enum ComColumnDefaultClass { COM_CURRENT_DEFAULT
+                           , COM_CURRENT_UT_DEFAULT
```

Comment:
@robertamarton Could you help to provide some comments here?
@selvaganesang  I think new values will NOT cause problems, the MD table structure is not changed. I don't find any checking code relying on values in this column. But Roberta should have more insight here.

Reply:
that is a good point. If a new value is being added to an existing enum in common/ComSmallDefs.h and that value will be stored in metadata, then it cannot be added in the middle of that enum. It should be added at the end.
Better yet, assign a number to  the enums so they dont cause an issue if a new
value is added in the middle.

Something like:
enum ComColumnDefaultClass { COM_CURRENT_DEFAULT                   = 0
                           , COM_NO_DEFAULT                        = 1
                           , COM_NULL_DEFAULT                      = 2
                           , COM_USER_DEFINED_DEFAULT              = 3
                           , COM_USER_FUNCTION_DEFAULT             = 4
                           , COM_IDENTITY_GENERATED_BY_DEFAULT     = 5
                           , COM_IDENTITY_GENERATED_ALWAYS         = 6
                           , COM_ALWAYS_COMPUTE_COMPUTED_COLUMN_DEFAULT = 7
                           , COM_ALWAYS_DEFAULT_COMPUTED_COLUMN_DEFAULT = 8
                           };

Reply:
I'm glad you found this. I would recommend to add a big comment saying that these values are stored in the metadata (COLUMNS.DEFAULT_CLASS) and cannot be changed (this is in addition to actually assigning the numbers, as Anoop said). There are only few such enums and it would be good if those were very clearly marked.

Reply:
Thanks all, I now understand what is the issue! If there are old metadata, and there is a upgrade, then things are messed up.
But I think this enum can be changed by adding new value at the end, instead of "cannot be changed". So let me add them at the end, and add big comments.